src/app/qgisapp.cpp

```diff
@@ -16745,23 +16752,40 @@ void QgisApp::read3DMapViewSettings( Qgs3DMapCanvasDockWidget *w, QDomElement &e
   }
   map->setOutputDpi( QgsApplication::desktop()->logicalDpiX() );
 
-  w->setMapSettings( map );
+  widget->setMapSettings( map );
 
   QDomElement elemCamera = elem3DMap.firstChildElement( QStringLiteral( "camera" ) );
   if ( !elemCamera.isNull() )
   {
-    w->mapCanvas3D()->cameraController()->readXml( elemCamera );
+    widget->mapCanvas3D()->cameraController()->readXml( elemCamera );
   }
 
   QDomElement elemAnimation = elem3DMap.firstChildElement( QStringLiteral( "animation3d" ) );
   if ( !elemAnimation.isNull() )
   {
     Qgs3DAnimationSettings animationSettings;
     animationSettings.readXml( elemAnimation );
-    w->animationWidget()->setAnimation( animationSettings );
+    widget->animationWidget()->setAnimation( animationSettings );
+  }
+
+  {
+    int dx = elem3DMap.attribute( QStringLiteral( "d_x" ), "0" ).toInt();
+    int dy = elem3DMap.attribute( QStringLiteral( "d_x" ), "0" ).toInt();
+    int dw = elem3DMap.attribute( QStringLiteral( "d_width" ), "200" ).toInt();
+    int dh = elem3DMap.attribute( QStringLiteral( "d_height" ), "200" ).toInt();
+    widget->dockableWidgetHelper()->setDialogGeometry( QRect( dx, dy, dw, dh ) );
   }
 
-  readDockWidgetSettings( w, elem3DMap );
+  {
+    int x = elem3DMap.attribute( QStringLiteral( "x" ), QStringLiteral( "0" ) ).toInt();
+    int y = elem3DMap.attribute( QStringLiteral( "y" ), QStringLiteral( "0" ) ).toInt();
+    int w = elem3DMap.attribute( QStringLiteral( "width" ), QStringLiteral( "200" ) ).toInt();
+    int h = elem3DMap.attribute( QStringLiteral( "height" ), QStringLiteral( "200" ) ).toInt();
+    bool floating = elem3DMap.attribute( QStringLiteral( "floating" ), QStringLiteral( "0" ) ).toInt();
+    Qt::DockWidgetArea area = static_cast< Qt::DockWidgetArea >( elem3DMap.attribute( QStringLiteral( "area" ), QString::number( Qt::RightDockWidgetArea ) ).toInt() );
+
+    widget->dockableWidgetHelper()->setDockGeometry( QRect( x, y, w, h ), floating, area );
```

Comment:
Like above, all this should go to a call `widget->dockableWidgetHelper()->readXml( elem3DMap );`

And thanks to that, we can probably also remove all the setDialogGeometry(), setDockGeometry(), dockGeometry(), dialogGeometry() calls from the helper class API to make it cleaner